### ISSUE TITLE
improve the Isitdown module

### ISSUE THREAD

**Issue by sh1ftsh** [leon-ai/leon#25]
### Expected Behavior
Be able to query this the status of some website without the fully qualified domain name.

### Actual Behavior
Currently a query like this fails
`is gitub up ?`
I should ask `is github.com up ?`

### Proposal
May be list some of the most popular websites and treat them with special query as I made just before

**Comment by louistiti** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [🔧 Improve Is It Down module with a given list of popular websites](https://trello.com/c/e1LU9xFe/227-%F0%9F%94%A7-improve-is-it-down-module-with-a-given-list-of-popular-websites)

**Comment by louistiti** [comment]
Hello @sh1ftsh,

Thanks for this suggestion.

This is something we could indeed do.

Is that something you would like to work on?

**Comment by adrijere** [comment]
I'm interested to do this one as first issue if you both agree.

**Comment by louistiti** [comment]
Hello @adrijere,

Yeah with pleasure, go for it :+1:

**Comment by sh1ftsh** [comment]
Hello, thanks the comment, for sure I want to work on this one too

**Comment by louistiti** [comment]
Okay cool! Please reference your PR to this issue once it's done 👍

**Comment by VMinhoto** [comment]
Hey! Can is this still open? Can I pick it up?

**Comment by louistiti** [comment]
Hello @VMinhoto,

The [module](https://github.com/leon-ai/leon/blob/develop/packages/checker/isitdown.py) is done but can be improved, feel free to do it :wink:

**Comment by VMinhoto** [comment]
I can try to improve that! The issue is:
Asking for 
```is gitub up ?``` instead of ```is github.com up?``` right?

**Comment by louistiti** [comment]
Indeed! And for the "most popular" ones.

**Comment by VMinhoto** [comment]
I see several ways to accomplish this. Will list them with cost/benefit explanation and could you guys select one depending on what you really want implemented:

- **Have a list of all websites registered and search them for the required string** - Would not advise this as there are billions of billions websites domain listed and it probable wold cause memory issues (and probably would take a lot of time to search on such a huge list of texts)
- **Have a list of possible domains (.com, .org, etc.) and iterate through them** - Problem with this would be country specific domains. Ping all possible combinations of the target string and the .xxx would probably take too much time.
- **Hybrid Solution** - Ping the most popular .xxx (.org, .com, .edu, etc.) and if it not finds it there ask the user for more information.

In my opinion the Hybrid Solution would be the best one, but what do you guys think about this?

**Comment by louistiti** [comment]
Thanks for putting it here.

The hybrid solution looks good, however we cannot ensure that the domain name is the right one in that way. Two websites might have the same name and then one could have `xxx.com` and another `xxx.io`.

So my suggestion here is to create a list of the most popular ones and do a mapping of it, in case the list start to grow quite well, then we could think of optimization (alphabetic indexing, etc.).
If it is not found, then we could fallback into the hybrid solution.

Also please note that this list must be part of the module config file, so it is easy for everyone to edit it and let fellows add their own favorite mappings.

What do you think?

**Comment by VMinhoto** [comment]
Hey again several more questions. Sry about the interrogations, but trying to get the 'requirements' before actually start implementing :).

- When you say ```list of the most popular ones``` are talking about a list of popular full websites? Like:
```
github.com
google.com
amazon.com
...
```
- Yes it makes sense that the list should be at the config file yes!

- Yes if domain name is provided as input it should maintain the current behavior.

**Comment by louistiti** [comment]
No problem, don't hesitate to ask :smile: 

Yes, I'm talking about a mapping like so: `github => github.com; google => google.com`, etc.

**Comment by louistiti** [comment]
Always take the main one, in the case of Amazon, it is indeed the `.com` TLD. Most of the popular website will take over from their and bind the right servers under the hood.

**Comment by VMinhoto** [comment]
Done for now. Will read a bit more on documentation and get it done!

**Comment by VMinhoto** [comment]
It's pretty much done the searching part, however I'm having trouble recognizing it has an entity. The main problems I am having are:

1. Failing to run package 

- If asked: ```Is google.com up?``` it answers: ```I am now requesting Google. Google is running correctly.```

- However if asked: ```Is google up?``` it doesn't recognize google as an entity, therefor it doesn't even run the package. It just answers ```Good afternoon!```

2. Multiple queuing

- If asked ```Check if github.com and nodejs.org are up``` it correctly answers: ``` I'm trying to reach Github. Github is running correctly. I am now requesting Nodejs. Nodejs is up.```
- But if asked ```Check if github and nodejs are up``` it answeres: ```Please provide me at least one valid domain name.```. It fails to recognize any identitity, even the question ```Check if github is up``` is working correctly.

Is there any way i can manipulate entity recognition to recognize several entities in the same setence?

**Comment by jSAMUELj** [comment]
Hi is this issue still free ? I would like to try working on it as my first issue.

**Comment by Divlo** [comment]
Yes, this issue is still free! Feel free to do it! Thanks for asking @jSAMUELj!
